.github/workflows/dotnet.yml

```
@@ -24,3 +24,18 @@ jobs:
       run: dotnet restore
     - name: Build
       run: dotnet build --no-restore
+    - name: Run Test
+      run: |
+         dotnet test --no-build --verbosity normal \
+            --logger "trx;LogFileName=test-results.trx" \
+            --collect:"XPlat Code Coverage" \
+            --settings .runsettings \
+            -- RunConfiguration.MaxCpuCount=1 TestRunParameters.Retry=3
+    - name: Publish Test Results
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)